### PR TITLE
Remove RHEL 6 support and procedures

### PR DESCRIPTION
**Pull request by Lennonka**
#### What changes are you introducing?

Removing RHEL 6 support and procedures from Sat 6.16 and later

#### Why are you introducing these changes? (Explanation, links to references, issues, etc.)

[SAT-33476](https://issues.redhat.com/browse/SAT-33476) (public)

#### Anything else to add? (Considerations, potential downsides, alternative solutions you have explored, etc.)

- In some cases, I wasn't entirely sure what to remove and what to keep.

#### Checklists

* [x] I am okay with my commits getting squashed when you merge this PR.
* [x] I am familiar with the [contributing](https://github.com/theforeman/foreman-documentation/blob/master/CONTRIBUTING.md) guidelines.

Please cherry-pick my commits into:

* [x] Foreman 3.14/Katello 4.16
* [x] Foreman 3.13/Katello 4.15 (EL9 only)
* [x] Foreman 3.12/Katello 4.14 (Satellite 6.16; orcharhino 7.2 on EL9 only)
* [ ] Foreman 3.11/Katello 4.13 (orcharhino 6.11 on EL8 only; orcharhino 7.0 on EL8+EL9; orcharhino 7.1 with Leapp)
* [ ] Foreman 3.10/Katello 4.12
* [ ] Foreman 3.9/Katello 4.11 (Satellite 6.15; orcharhino 6.8/6.9/6.10)
* [ ] Foreman 3.8/Katello 4.10
* [ ] Foreman 3.7/Katello 4.9 (Satellite 6.14)
* We do not accept PRs for Foreman older than 3.7.

### CODE REVIEW

File: guides/common/modules/proc_using-tftp-services-through-nat.adoc [hunk 1]
```diff
@@ -18,12 +18,9 @@ endif::[]
 ----
 include::snip_make-firewall-settings-persistent.adoc[]
 
-ifdef::satellite[]
-.Using NAT on {RHEL} 6:
-endif::[]
 ifndef::satellite[]
 .Using NAT on linux with `iptables`:
```
Comment (Lennonka) [inline guides/common/modules/proc_using-tftp-services-through-nat.adoc:22]: Should I remove this whole procedure with `iptables`?

Comment (ekohl) [reply guides/common/modules/proc_using-tftp-services-through-nat.adoc:22]: I'm questioning the whole procedure. Already did so when looking at the new provisioning infrastructure services guide, but didn't want to blow up the scope.

I don't quite know what the point of the whole chapter is. I'd be tempted to just drop it altogether so I've opened https://github.com/theforeman/foreman-documentation/pull/3852.

File: guides/common/modules/proc_creating-custom-rhel-images.adoc [hunk 2]
```diff
@@ -6,7 +6,7 @@
 In this example, we use a {RHEL} 7 Workstation.
```
Comment (ekohl) [inline guides/common/modules/proc_creating-custom-rhel-images.adoc:6]: We should really update this to use RHEL 9.

File: guides/common/modules/proc_configuring-a-host-for-registration.adoc [hunk 3]
```diff
@@ -23,13 +23,6 @@ ifdef::client-content-dnf[]
 ----
```
Comment (ekohl) [inline guides/common/modules/proc_configuring-a-host-for-registration.adoc:23]: Just above this is the "On {EL} 7 and later" header. Is that header still relevant now or can we drop it instead?

File: guides/common/modules/con_smartproxy-performance-tests.adoc [hunk 4]
```diff
@@ -41,5 +41,5 @@ We have tested executing Remote Execution jobs via both SSH and Ansible backend
 All configurations were able to handle all of the tests without errors, except for the smallest configuration (4 CPUs and 12 GiB memory) which failed to finish on all 4000 hosts.
 
 .Content sync use case
-In a sync test where we synced {RHEL} 6, 7, 8 BaseOS and 8 AppStream we have not seen significant differences among {SmartProxy} configurations.
+In a sync test where we synced {RHEL} 7, 8 BaseOS and 8 AppStream we have not seen significant differences among {SmartProxy} configurations.
```
Comment (ekohl) [inline guides/common/modules/con_smartproxy-performance-tests.adoc:44]: I think this is a bit of an edge case. It describes a specific experiment performed, not the supported OS. How relevant that actually is can be debated. IMHO the whole chapter paragraph doesn't say a lot and I'm tempted to drop it.

**Review by ekohl — COMMENTED** [review]
I'll be glad to see `bootstrap.py` go away.

File: guides/common/modules/proc_configuring-a-host-for-registration.adoc (hunk 3 above)
Comment (maximiliankolb) [reply guides/common/modules/proc_configuring-a-host-for-registration.adoc:23]: I suggest to reword line 21 to "On {EL}:". What do you think Lena?

File: guides/common/modules/proc_creating-custom-rhel-images.adoc (hunk 2 above)
Comment (Lennonka) [reply guides/common/modules/proc_creating-custom-rhel-images.adoc:6]: Please, really file a Jira issue.

**Review by ekohl — APPROVED** [review]
